### PR TITLE
Make route refresh check more required options

### PR DESCRIPTION
**Pull request by kmadsen**
## Description

Route refresh returns errors if the overview is not full. Add this check to enable.

- [ ] I have added any issue links
- [x] I have added all related labels (`bug`, `feature`, `new API(s)`, `SEMVER`, etc.)
- [ ] I have added the appropriate milestone and project boards

## Testing

Please describe the manual tests that you ran to verify your changes

- [ ] I have tested locally (including `SNAPSHOT` upstream dependencies if needed) through testapp/demo app and run all activities to avoid regressions
- [x] I have tested via a test drive, or a simulation/mock location app
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have updated the `CHANGELOG` including this PR
- [ ] We might need to update / push `api/current.txt` files after running `$> make 1.0-core-update-api` (Core) / `$> make 1.0-ui-update-api` (UI) if there are changes / errors we're 🆗 with (e.g. `AddedMethod` changes are marked as errors but don't break SemVer) 🚀 If there are SemVer breaking changes add the `SEMVER` label. See [Metalava](https://github.com/mapbox/mapbox-navigation-android/blob/master/docs/metalava.md) docs

### CODE REVIEW

File: libnavigation-core/src/main/java/com/mapbox/navigation/core/routerefresh/RouteRefreshController.kt [hunk 1]
```diff
@@ -53,8 +53,10 @@ internal class RouteRefreshController(
 
     private fun supportsRefresh(route: DirectionsRoute?): Boolean {
         val isTrafficProfile = route?.routeOptions()
-            ?.profile()?.equals(DirectionsCriteria.PROFILE_DRIVING_TRAFFIC)
-        return isTrafficProfile == true
+            ?.profile()?.equals(DirectionsCriteria.PROFILE_DRIVING_TRAFFIC) ?: false
+        val isRouteOverviewFull = route?.routeOptions()
+            ?.overview()?.equals(DirectionsCriteria.OVERVIEW_FULL) ?: false
+        return isTrafficProfile && isRouteOverviewFull
```
Comment (danpaz) [inline libnavigation-core/src/main/java/com/mapbox/navigation/core/routerefresh/RouteRefreshController.kt:59]: Note that either congestion or maxspeed annotations also need to be set for refresh to be supported.

Comment (kmadsen) [reply libnavigation-core/src/main/java/com/mapbox/navigation/core/routerefresh/RouteRefreshController.kt]: ah ok cool, added another check

**Comment by danpaz** [comment]
👍  LGTM

**Comment by kmadsen** [comment]
> Could we add a Unit Test to cover the logic just added @ kmadsen 🙏?

amazing tests added 👍

**Pull request merged**